index.js

```diff
@@ -18,20 +18,27 @@ const getIndexMapping = index => indexMappings[index] || '';
 const recodeSearchQuery = query => query.split(/\+|\s/).join("%20");
 
 const getQueryFromParams = (url, parsedURL) => {
-  console.log('getting query from params', parsedURL);
-  // const searchArg = url.match(/\?.*searcharg=([^&]+)/);
   const searchArg = parsedURL.searchParams.get('searcharg');
-  console.log('searchArg: ', searchArg);
-  // const searchArgAlt = url.match(/\?.*SEARCH=([^&]+)/);
   const searchArgAlt = parsedURL.searchParams.get('SEARCH');
-  console.log('searchArgAlt: ', searchArgAlt);
-  // const q = searchArg ? searchArg[1] : (searchArgAlt ? searchArgAlt[1] : '');
-  const q = searchArg || searchArgAlt || '';
-  console.log('q: ', q);
-  // const searchType = url.match(/\?.*searchtype=([^&]+)/);
-  const searchType = parsedURL.searchParams.get('searchtype');
-  console.log('searchType: ', searchType);
-  // return recodeSearchQuery(q + (searchType ? getIndexMapping(searchType[1]) : ''));
+  let searchIndex;
+  let searchArgFromQueryParam;
+  let breaker = false;
+  parsedURL.searchParams.forEach((v,k) => {
+    if (!v && !breaker) {
```

Comment:
This could use a comment

Reply:
Actually I think this whole bit from line 26 should be in a function/method taking `parseURL.searchParams` as an argument. Then you can just say 
`const searchArgFromQueryParam = descriptiveFunctionName(parseURL.searchParams)`